ambition_utils/rrule/models.py

```diff
@@ -304,3 +351,44 @@ def generate_dates_from_params(cls, rrule_params, time_zone=None, num_dates=20):
         rule = cls(rrule_params=rrule_params, time_zone=time_zone)
 
         return rule.generate_dates(num_dates=num_dates)
+
+    @classmethod
+    def clone(cls, source: RRule) -> RRule:
+        """
+        Creates a clone of a passed RRule object offset.
+        """
+
+        # Create a clone of the source RRule.
+        # Clear id to force a new object.
+        clone = copy.deepcopy(source)
+        clone.id = None
+
+        clone.save()
+
+        return clone
+
+    @classmethod
+    def clone_with_offset(cls, source: RRule, day_offset: int) -> RRule:
+        """
+        Creates a clone of a passed RRule object offset by a specified number of days.
+        Days can be negative.
+        """
+
+        # Create a clone of the source RRule.
+        clone = cls.clone(source)
+
+        # Manually update the rrule.dtstart & next_occurrence with the offset.
+        clone.rrule_params['dtstart'] = parser.parse(clone.rrule_params['dtstart']) + timedelta(days=day_offset)
+        clone.next_occurrence = clone.next_occurrence + timedelta(days=day_offset)
+
+        # Update byweekday params by offsetting each one present.
+        if 'byweekday' in clone.rrule_params:
```

Comment:
if this method is assuming a day offset, i'd rather name it clone_with_day_offset.

otherwise if we plan to support other offsets like week, month, etc, i think the day_offset arg should be an optional kwarg (`days=None`) and match the delta kwargs.

also, the timedelta might as well be relativedelta if we do plan to support other kwargs.

if we do plan to support other kwargs, we can handle that as we need it, but i'd at least like to name the kwargs in a way that we won't have to make changes later. my vote would be to use the naming convention as relativedelta / fleming and handle each case in its own method.

for example assuming days=5 is passed
```
if days:
    clone.rrule_params = self.apply_offset_days(clone.rrule_params, days)
```    
or something like that. just trying to think of a way to break out the logic so this one method doesn't get more complex any time we add additional logic. also it might make it easier to test this specific part of the cloning for different types of offsets.

Reply:
I like the vision for that flexibility but I can't imagine us giving users that much flexibility in choosing offsets. I have renamed it based on your suggestion, 1b4b60a, but would like to not add more functionality at this time. 

Would you rather see the flexible kwargs now and throw exceptions if something other than `days` is passed? I don't feel like we gain much by having a consistent signature if we want it to be more flexible.